src/AppBundle/Controller/UserController.php

```diff
@@ -353,6 +353,14 @@ private function populateUser(EntityDir\User $user, array $data)
             $user->setLastLoggedIn(new \DateTime($data['last_logged_in']));
         }
 
+        if (!empty($data['pa_team_name'])) {
+            if ($user->getTeams()->isEmpty()) {
+                $user->addTeam(new EntityDir\Team());
+            }
+            $team = $user->getTeams()->first()->setTeamName($data['pa_team_name']);
+            $this->persistAndFlush($team);
+        }
```

Comment:
Id pass the team name as a constructor argument to the Team entity (and make it optional if there is a use case for having teams with no name). Keeps business logic in api layer.